### ISSUE TITLE
Better contrib packages naming consistency

### ISSUE THREAD

**Issue by rth** [scikit-learn-contrib/scikit-learn-contrib#27]
The package naming is also something that might be worth checking during the review to accept into `scikit-learn-contrib`. In particular, to ensure that,
  * the github repo name
  * the uploaded name on PyPi
  * the Python package name when it's imported

are somewhat consistent (or follow a similar logic).

For instance, 

1. for pyearth,
  * `scikit-learn-contrib/py-earth`
  * `pip install scikit-learn-contrib-py-earth`
  * `import pyearth`

2. for boruta_py,
  * `scikit-learn-contrib/boruta_py`
  * `pip install Buruta`
  * `import boruta`

3. for forest-confidence-interval
 * `scikit-learn-contrib/forest-confidence-interval`
 * `pip install forestci`
 * `import forestci`

Other currently included packages are mostly consistent in this respect, I think.

I'm not saying that all 3 should be identical but some more consistency (how to handle the "py" prefix/suffix, underscores / dash) wouldn't hurt. Also at least the github repo and the pypi package should, IMO, match. The github repo name is probably the easiest to adapt, as Github will do automatic redirects.

**Comment by mblondel** [comment]
I agree on principle that more consistency would be nice but many projects existed before moving to contrib. It would be difficult to require changing the package name, pip name etc.

**Comment by rth** [comment]
I agree. That's why the Github repository name is still the most easily changeable part. First because of redirects and then because when included here the url changes anyway.. Say packages 

Actually, I noticed that the "vision" section of the Readme has,
> With scikit-learn-contrib, users can install a project by pip install sklearn-contrib-project-name

which does makes sense : same as  `sklearn-contrib/project-name` (Github) and I imagine makes analogies with `from scikit.learn.contrib import package` (when scikit-learn was a scikit). 

But at the same time, a lot of included packages don't do this. Also `pip install hdbscan` is just so much nicer than `pip install sklearn-contrib-hdbscan` because the latter name  puts more emphasis on "sklearn" than on "hdbscan".   

Also for more complicated names that doesn't work so well `sklearn-contrib-imbalanced-learn`, `sklearn-contrib-forest-confidence-interval`.
 I'm looking forward to when `sklearn-contrib-t-Student-Mixture-Models` (https://github.com/scikit-learn-contrib/scikit-learn-contrib/issues/18) is accepted :)

So maybe it could be worth re-evaluating the naming, and write it down somewhere, at least for the ideal case scenario?

 For instance, this could be,
  * Name: t-Student-Mixtures (sklearn-contrib)
  * `pip install t-student-mixtures`
  * "sklearn-contrib/t-student-mixtures"
  * `import tstudent_mixtures` (if possible or `smm` as it currently is)

this would imply asking to change the name occasionally, but at the same time I imagine projects that get accepted into the Apache incubator also change names occasionally (need to check that) and sometimes it's for the best..
